tests/roxctl/bats-tests/cluster/scanner-generate.bats

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bats
+
+load "../helpers.bash"
+
+output_dir=""
+
+setup_file() {
+  local -r roxctl_version="$(roxctl-development version || true)"
+  echo "Testing roxctl version: '${roxctl_version}'" >&3
+
+  command -v cat || skip "Command 'cat' required."
+  command -v grep || skip "Command 'grep' required."
+  [[ -n "${API_ENDPOINT}" ]] || fail "Environment variable 'API_ENDPOINT' required"
+  [[ -n "${ROX_PASSWORD}" ]] || fail "Environment variable 'ROX_PASSWORD' required"
+}
+
+setup() {
+  output_dir="$(mktemp -d -u)"
+}
+
+teardown() {
+  rm -rf "${output_dir}"
+}
+
+scanner_generate() {
+  run roxctl_authenticated scanner generate \
+    --output-dir "${output_dir}" "$@"
+
+  assert_success
+}
+
+run_scanner_generate_and_check() {
+  local -r cluster_type="$1";shift
+
+  scanner_generate --cluster-type "${cluster_type}" "$@"
+  assert_success
+}
+
+assert_number_of_k8s_resources() {
+    local -r k8s_resources_count=$(cat "${output_dir}/scanner/"*.yaml | grep -c "^apiVersion") || true
+
+    [[ "${k8s_resources_count}" = "${1}" ]] || fail "Unexpected number of k8s resources"
+}
+
+@test "[openshift4] roxctl scanner generate" {
+  run_scanner_generate_and_check openshift4
+
+  assert_file_exist "${output_dir}/scanner/02-scanner-06-deployment.yaml"
+
+  # The only difference between OpenShift and OpenShift 4 configurations is that OpenShift 4 configuration has additional
+  # mounted volume. It's called "trusted-ca-volume" and we use it to identify OpenShift 4 configuration.
+  run -0 grep -q 'trusted-ca-volume' "${output_dir}/scanner/02-scanner-06-deployment.yaml"
+  assert_number_of_k8s_resources 16
+}
+
+@test "[openshift] roxctl scanner generate" {
+  run_scanner_generate_and_check openshift
+
+  assert_file_exist "${output_dir}/scanner/02-scanner-06-deployment.yaml"
+  run -0 grep -q 'ROX_OPENSHIFT_API' "${output_dir}/scanner/02-scanner-06-deployment.yaml"
```

Comment:
does this check for the existence of `ROX_OPENSHIFT_API` env var? If so, how come we do not check in OpenShift4?

Reply:
@RTann Yes, it checks for the `ROX_OPENSHIFT_API` environment variable and that variable exists for OpenShift4 too, but OpenShift4 can be uniquely identified by checking for additional volume - in our case `trusted-ca-volume`. Additional check for `ROX_OPENSHIFT_API` in the case of OpenShift4 would not be beneficial.

Reply:
What exactly are we checking for? For example, I'd expect `ROX_OPENSHIFT_API` to exist in both OpenShift 3.11 and OpenShift 4.x, and it would be an error if it did not exist

Reply:
@RTann, I have also added a check for `ROX_OPENSHIFT_API` in the OS4 test.